src/tools/dhcpv6-discover.c

```diff
@@ -0,0 +1,823 @@
+/* Pi-hole: A black hole for Internet advertisements
+*  (c) 2024 Pi-hole, LLC (https://pi-hole.net)
+*  Network-wide ad blocking via your own hardware.
+*
+*  FTL Engine
+*  DHCPv6 / ICMPv6 discovery routines
+*
+*  Inspired by the ndisc6 project
+*
+*  This file is copyright under the latest version of the EUPL.
+*  Please see LICENSE file for your rights under this license. */
+
+#include "dhcpv6-discover.h"
+#include "dhcp-discover.h"
+
+// check_capability()
+#include "capabilities.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <errno.h>
+#include <unistd.h>
+#include <time.h>
+#include <poll.h>
+#include <sys/socket.h>
+#include <fcntl.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <net/if.h>
+#include <netinet/in.h>
+#include <netinet/icmp6.h>
+
+/**
+ * @brief Resolves an IPv6 address by hostname and interface name.
+ *
+ * This function takes a hostname and an interface name, resolves the hostname
+ * to an IPv6 address, and fills the provided sockaddr_in6 structure with the
+ * resolved address.
+ *
+ * @param name The hostname to resolve.
+ * @param ifname The name of the network interface.
+ * @param numeric A flag indicating whether the hostname is a numeric IPv6 address.
```

Comment:
This param does not exist in the function call. Only three arguments are taken.

Reply:
Ah, right. This parameter *did exist* but I removed it later because I made this the default as it is often not useful to try to reverse-resolve an IPv6 address in internal networks and it was causing some long delays because (at least my ISP's) DNS servers time can run into timeouts when trying to resolve GUA IPv6 addresses.

Reply:
Let's make this a suggestions here we can later collect in case you have further suggestions (avoiding tiny commits):
```suggestion
```